### PR TITLE
Comp 305 publish json schema for iqm circuit execution request

### PR DESCRIPTION
**Pull request by q-mat-beu**
Add capability to publish a json schema of the circuit execution request that is sent to the IQM backend during the docs build step.

### CODE REVIEW

**Review by smite — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+        with: 
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (q-mat-beu) [reply .github/workflows/ci.yml:19]: Yes we need the git history for fetching the latest tag.

File: docs/generate_json_schemas.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    return str(tags[-1])
+
+def generate_json_schema(cls: type[BaseModel], filename: str) -> dict[str, Any]:
+    """Generate a JSON schema dictionary from the given Pydantic model.
+    """
+    json_schema = cls.schema()
+    tag = _get_git_tag()
+    return {
+        # JSON Schema version used
+        '$schema': 'https://json-schema.org/draft/2020-12/schema',
```
Comment (hay-k) [inline docs/generate_json_schemas.py:44]: Maybe a naive question, but why does this have `draft` in the path?

File: docs/generate_json_schemas.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+
+def _get_git_tag() -> str:
+    repo = Repo(os.getcwd())
+    tags = sorted(repo.tags, key=lambda t: t.commit.committed_datetime)
+    return str(tags[-1])
+
+def generate_json_schema(cls: type[BaseModel], filename: str) -> dict[str, Any]:
+    """Generate a JSON schema dictionary from the given Pydantic model.
```
Comment (hay-k) [inline docs/generate_json_schemas.py:38]: The arguments are not documented in the docstring.

File: docs/generate_json_schemas.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    """
+    json_schema = cls.schema()
+    tag = _get_git_tag()
+    return {
+        # JSON Schema version used
+        '$schema': 'https://json-schema.org/draft/2020-12/schema',
+        # URI containing the version of the generated schema
+        '$id': f'https://www.meetiqm.com/iqm_client/{filename}_v{tag}.json',
```
Comment (hay-k) [inline docs/generate_json_schemas.py:46]: Is this a valid URL though?

File: docs/generate_json_schemas.py (hunk 2 above)
Comment (q-mat-beu) [reply docs/generate_json_schemas.py:44]: I think each JSON schema standard is a draft. See here: https://json-schema.org/specification-links.html#2020-12

File: docs/generate_json_schemas.py (hunk 4 above)
Comment (q-mat-beu) [reply docs/generate_json_schemas.py]: Now it should be.

**Review by hay-k — APPROVED** [review]

**Pull request merged**